From c9106b041cf644df468162f5f15ae384df6c538e Mon Sep 17 00:00:00 2001
From: Arfrever Frehtes Taifersar Arahesis <Arfrever@Apache.Org>
Date: Tue, 17 Nov 2020 00:00:00 +0000
Subject: [PATCH 1/2] Fix parallel build failure (#322)

Add separate custom target for copying of libopencc library to directory
of opencc_dict.

Signed-off-by: Arfrever Frehtes Taifersar Arahesis <Arfrever@Apache.Org>
---
 data/CMakeLists.txt | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/data/CMakeLists.txt b/data/CMakeLists.txt
index ae1917b..e9b921f 100644
--- a/data/CMakeLists.txt
+++ b/data/CMakeLists.txt
@@ -116,14 +116,25 @@ foreach(DICT ${DICTS_GENERATED})
   )
 endforeach(DICT)
 
+add_custom_target(
+  copy_libopencc_to_dir_of_opencc_dict
+  COMMENT
+    "Copying libopencc to directory of opencc_dict"
+  COMMAND
+    ${CMAKE_COMMAND} -E copy "$<TARGET_FILE:libopencc>" "$<TARGET_FILE_DIR:${OPENCC_DICT_BIN}>"
+)
+if (WIN32)
+  set(DICT_WIN32_DEPENDS copy_libopencc_to_dir_of_opencc_dict)
+else()
+  set(DICT_WIN32_DEPENDS)
+endif()
+
 foreach(DICT ${DICTS})
   add_custom_command(
     OUTPUT
       ${DICT}.ocd2
     COMMENT
       "Building ${DICT}.ocd2"
-    COMMAND
-      ${CMAKE_COMMAND} -E copy "$<TARGET_FILE:libopencc>" "$<TARGET_FILE_DIR:${OPENCC_DICT_BIN}>"
     COMMAND
       ${OPENCC_DICT_BIN}
         --input ${DICT_${DICT}_INPUT}
@@ -131,6 +142,7 @@ foreach(DICT ${DICTS})
         --from text
         --to ocd2
     DEPENDS
+      ${DICT_WIN32_DEPENDS}
       ${OPENCC_DICT_BIN}
       ${DICT_${DICT}_INPUT}
   )
-- 
2.29.2


From 4804bdfabf888d8ae087461f350ad2a4791b0576 Mon Sep 17 00:00:00 2001
From: Arfrever Frehtes Taifersar Arahesis <Arfrever@Apache.Org>
Date: Fri, 18 Dec 2020 00:00:00 +0000
Subject: [PATCH 2/2] Improve descriptions of some actions in build system.

Signed-off-by: Arfrever Frehtes Taifersar Arahesis <Arfrever@Apache.Org>
---
 src/CMakeLists.txt           | 4 ++--
 src/benchmark/CMakeLists.txt | 8 ++++----
 test/CMakeLists.txt          | 4 ++--
 3 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 4883990..b184fae 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -140,12 +140,12 @@ if (ENABLE_GTEST)
     add_custom_target(
         copy_gtest_to_src
         ${CMAKE_COMMAND} -E copy $<TARGET_FILE:gtest> ${CMAKE_CURRENT_BINARY_DIR}
-      COMMENT "Copy gtest"
+      COMMENT "Copying gtest to src"
     )
     add_custom_target(
         copy_gtest_main_to_src
         ${CMAKE_COMMAND} -E copy $<TARGET_FILE:gtest_main> ${CMAKE_CURRENT_BINARY_DIR}
-      COMMENT "Copy gtest_main"
+      COMMENT "Copying gtest_main to src"
     )
   endif()
 
diff --git a/src/benchmark/CMakeLists.txt b/src/benchmark/CMakeLists.txt
index a3f1ae7..25a1d1f 100644
--- a/src/benchmark/CMakeLists.txt
+++ b/src/benchmark/CMakeLists.txt
@@ -8,12 +8,12 @@ if (WIN32)
   add_custom_target(
     copy_benchmark
     ${CMAKE_COMMAND} -E copy $<TARGET_FILE:benchmark> ${CMAKE_CURRENT_BINARY_DIR}
-    COMMENT "Copy benchmark"
+    COMMENT "Copying benchmark to src/benchmark"
   )
   add_custom_target(
-    copy_opencc
+    copy_libopencc
     ${CMAKE_COMMAND} -E copy $<TARGET_FILE:libopencc> ${CMAKE_CURRENT_BINARY_DIR}
-    COMMENT "Copy opencc"
+    COMMENT "Copying libopencc to src/benchmark"
   )
-  add_dependencies(performance copy_benchmark copy_opencc)
+  add_dependencies(performance copy_benchmark copy_libopencc)
 endif()
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index d365960..7a61342 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -18,12 +18,12 @@ if (ENABLE_GTEST)
     add_custom_target(
         copy_gtest_to_test
         ${CMAKE_COMMAND} -E copy $<TARGET_FILE:gtest> ${CMAKE_CURRENT_BINARY_DIR}
-      COMMENT "Copy gtest"
+      COMMENT "Copying gtest to test"
     )
     add_custom_target(
         copy_gtest_main_to_test
         ${CMAKE_COMMAND} -E copy $<TARGET_FILE:gtest_main> ${CMAKE_CURRENT_BINARY_DIR}
-      COMMENT "Copy gtest_main"
+      COMMENT "Copying gtest_main to test"
     )
   endif()
 
-- 
2.29.2
